zmon_worker_extras/check_plugins/exasol.py

```
@@ -4,89 +4,81 @@
         self._out = None
         self.user = user
         self.__password = password
         self.cluster = cluster
         self.schema = schema
-        self.java_opts = ['-Djava.net.preferIPv4Stack=true', '-Djava.awt.headless=true', '-Xmx512m', '-Xms128m']
-        self.exaplus_opts = [
-            '-recoverConnection',
-            'OFF',
-            '-retry',
-            '0',
-            '-lang',
-            'EN',
-            '-q',
-            '-x',
-            '-Q',
-            '10',
-            '-pipe',
-        ]
-        self.jar_file = '/server/exasol/exaplus/current/exaplus.jar'
+        self.driver = driver
 
     def query(self, query):
-        fd, name = tempfile.mkstemp(suffix='.sql', text=True)
         try:
-            fh = os.fdopen(fd, 'w')
-            fh.write('%s\n' % query)
-            fh.flush()
-
-            cmd = ['/usr/bin/java']
-            cmd.extend(self.java_opts)
-            cmd.extend(['-jar', self.jar_file])
-            cmd.extend(['-c', self.cluster])
-            cmd.extend(['-u', self.user])
-            cmd.extend(['-p', self.__password])
-            cmd.extend(self.exaplus_opts)
-            if self.schema:
-                cmd.extend(['-s', self.schema])
-            cmd.extend(['-f', name])
-            # print "EXAPLUS="+" ".join(cmd)
-            sub = subprocess.Popen(cmd, shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE, close_fds=True)
-            d_out, d_err = sub.communicate()
-            self._out = d_out
-            self._err = d_err
-        finally:
-            os.unlink(name)
+            connection = connect(
+                driver=self.driver,
+                EXAHOST=self.cluster,
+                EXASCHEMA=self.schema,
+                EXAUID=self.user,
+                EXAPWD=self.__password)
+            cursor = connection.cursor()
+
+            cursor.execute(query)
+            self._out = cursor
+
+        except Exception as err:
+            self._err = err
```

I suggest dropping the return ``tuple`` interface. If there is an exception then we either let it raise and bubble or we catch (filter exceptions?) and raise ``CheckError`` with a more informative message. We only return the result in the end.